plugin/trino-hive-hadoop2/src/test/java/io/trino/plugin/hive/AbstractTestHiveFileSystemS3.java

```diff
@@ -93,4 +132,134 @@ public void testIgnoreHadoopFolderMarker()
+        createFile(writableBucket, format("%s/part=simple/plain-file.txt", basePrefix));
+        createFile(writableBucket, format("%s/part=nested/parent/_nested-hidden-file.txt", basePrefix));
+        createFile(writableBucket, format("%s/part=nested/parent/nested-file.txt", basePrefix));
+        createFile(writableBucket, format("%s/part=plus+sign/plus-file.txt", basePrefix));
+        createFile(writableBucket, format("%s/part=percent%%sign/percent-file.txt", basePrefix));
+        createFile(writableBucket, format("%s/part=url%%20encoded/url-encoded-file.txt", basePrefix));
+        createFile(writableBucket, format("%s/part=level1|level2/pipe-file.txt", basePrefix));
+        createFile(writableBucket, format("%s/part=level1|level2/parent1/parent2/nested-file.txt", basePrefix));
+        createDirectory(writableBucket, format("%s/empty-directory/", basePrefix));
+        createFile(writableBucket, format("%s/.hidden-in-base.txt", basePrefix));
+
+        // List recursively through hive file iterator
+        HiveFileIterator recursiveIterator = new HiveFileIterator(
+                fakeTable,
+                Location.of(basePath.toString()),
+                trinoFileSystem,
+                new FileSystemDirectoryLister(),
+                new HdfsNamenodeStats(),
+                HiveFileIterator.NestedDirectoryPolicy.RECURSE);
+
+        List<String> recursiveListing = Streams.stream(recursiveIterator)
+                .map(TrinoFileStatus::getPath)
+                .toList();
+        // Should not include directories, or files underneath hidden directories
+        assertEqualsIgnoreOrder(
+                recursiveListing,
+                ImmutableList.of(
+                        format("%s/part=simple/plain-file.txt", basePath),
+                        format("%s/part=nested/parent/nested-file.txt", basePath),
+                        format("%s/part=plus+sign/plus-file.txt", basePath),
+                        format("%s/part=percent%%sign/percent-file.txt", basePath),
+                        format("%s/part=url%%20encoded/url-encoded-file.txt", basePath),
+                        format("%s/part=level1|level2/pipe-file.txt", basePath),
+                        format("%s/part=level1|level2/parent1/parent2/nested-file.txt", basePath)));
+
+        HiveFileIterator shallowIterator = new HiveFileIterator(
+                fakeTable,
+                Location.of(basePath.toString()),
+                trinoFileSystem,
+                new FileSystemDirectoryLister(),
+                new HdfsNamenodeStats(),
+                HiveFileIterator.NestedDirectoryPolicy.IGNORED);
+        List<Path> shallowListing = Streams.stream(shallowIterator)
+                .map(TrinoFileStatus::getPath)
+                .map(Path::new)
+                .toList();
+        // Should not include any hidden files, folders, or nested files
+        assertEqualsIgnoreOrder(shallowListing, ImmutableList.of());
```

Comment:
Maybe it would be good to have at least one non hidden file on this level to check if it's listed?

Reply:
In such a case, the file would pop up in the listing.

Reply:
Yes, the idea is to test listening on top level.